astroquery/dastcom5/core.py

```diff
@@ -0,0 +1,340 @@
+from __future__ import print_function
+import re
+import os
+import urllib.request
+import zipfile
+import numpy as np
+import pandas as pd
+from tqdm import tqdm
+
+from astropy.time import Time
+from astropy.table import Table
+import astropy.units as u
+
+from . import conf
+from ..query import BaseQuery
+from ..utils import async_to_sync
+
+
+@async_to_sync
+class Dastcom5Class(BaseQuery):
+    """
+    Class for querying NEOs orbit from DASTCOM5.
+    """
+
+    def __init__(self, spk_id=None, name=None):
+        """
+        Parameters:
+        ----------
+        api_key : str
+            NASA OPEN APIs key (default: `DEMO_KEY`)
+        """
+        super(Dastcom5Class, self).__init__()
+        self.spk_id = spk_id
+        self.name = name
+        self.local_path = conf.ASTROQUERY_LOCAL_PATH
+        self.dbs_path = conf.DBS_LOCAL_PATH
+        self.ast_path = conf.AST_DB_PATH
+        self.com_path = conf.COM_DB_PATH
+        self.com_dtype = conf.COM_DTYPE
+        self.ast_dtype = conf.AST_DTYPE
+        self.ftp_url = conf.FTP_DB_URL
+        self._show_download_progress = self._Show_Download_Progress
+
+    def download_dastcom5(self):
+        """Downloads DASTCOM5 database.
+
+        Downloads and unzip DASTCOM5 file in default astroquery path (~/.astroquery).
+
+        """
+
+        dastcom5_dir = os.path.join(self.local_path, "dastcom5")
+        dastcom5_zip_path = os.path.join(self.local_path, "dastcom5.zip")
+
+        if os.path.isdir(dastcom5_dir):
+            raise FileExistsError(
+                "dastcom5 is already created in " +
+                os.path.abspath(dastcom5_dir)
+            )
+        if not zipfile.is_zipfile(dastcom5_zip_path):
+            if not os.path.isdir(self.local_path):
+                os.makedirs(self.local_path)
+
+            print("Downloading datscom5.zip")
+            with self._show_download_progress(unit='B', unit_scale=True, miniters=1, desc="dastcom5.zip") as t:
+                urllib.request.urlretrieve(
+                    self.ftp_url + "dastcom5.zip", filename=dastcom5_zip_path, reporthook=t.update_to)
+
+        with zipfile.ZipFile(dastcom5_zip_path) as myzip:
+            myzip.extractall(self.local_path)
+
+    class _Show_Download_Progress(tqdm):
```

Comment:
I don't know what other astroquery developers think, but isn't it extremely weird to have a class here?

Reply:
Yes, particularly because we have already implemented this in `astroquery.query`

Reply:
Is it? Okay! Will figure it out somehow

Reply:
I removed the class and used the astroquery's default downloading function